### PR TITLE
feat: add limits

### PR DESCRIPTION
**Pull request by TacticalTechJay**
Along with limits, uploading text page now catches and shows you any error.

### CODE REVIEW

**Comment by TacticalTechJay** [comment]
for #164

**Comment by TacticalTechJay** [comment]
This is open for debate if it should even be included with Zipline.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -92,3 +93,13 @@ model Invite {
   createdBy   User @relation(fields: [createdById], references: [id])
   createdById Int
 }
+
+model Limit {
+  id         Int      @id @default(autoincrement())
+  type_time  String   @default("daily")
+  limit_by   String   @default("count")
```
Comment (ItzDerock) [inline prisma/schema.prisma]: I think using an enum for `type_time` and `limit_by` would be better than using String.

File: src/pages/api/auth/create.ts [hunk 2]
```diff
@@ -82,6 +84,27 @@ async function handler(req: NextApiReq, res: NextApiRes) {
 
   delete newUser.password;
 
+  if (limits) {
+    const data = {};
+    const ends = ['type_time', 'limit_by', 'limit'];
+    for (const [x, y] of Object.entries(limits)) {
+      if (ends.includes(x)) data[x] = [y][0];
+      else continue;
+    }
```
Comment (axisiscool) [inline src/pages/api/auth/create.ts]:
```suggestion
    for (const [x, y] of Object.entries(limits)) {
      if (ends.includes(x)) data[x] = [y][0];
    }
```
Useless jump statement. It'll continue forward regardless of whether or not the `if` check passes.

File: src/pages/api/user/[id].ts [hunk 3]
```diff
@@ -124,6 +124,41 @@ async function handler(req: NextApiReq, res: NextApiRes) {
...
+            },
+          },
+        });
+      } else {
+        const data = {};
+        const ends = ['type_time', 'limit_by', 'limit'];
+        for (const [x, y] of Object.entries(req.body.limits)) {
+          if (ends.includes(x)) data[x] = [y][0];
+          else continue;
+        }
```
Comment (axisiscool) [inline src/pages/api/user/[id].ts]:
```suggestion
        for (const [x, y] of Object.entries(req.body.limits)) {
          if (ends.includes(x)) data[x] = [y][0];
        }
```
Ditto

**Review by axisiscool — CHANGES_REQUESTED** [review]

**Comment by TacticalTechJay** [comment]
Limits are canceled 👏

**Pull request closed without merging**